src/content/en/2021/javascript.md

```diff
@@ -1,20 +1,1544 @@
+### Minification
+
+When working towards reducing the script parsing time, we tend to focus on all the opportunities to make our code smaller and relevant. One such idea is to minify the files and bring down the bytes usage.
+
+The lighthouse report also [highlights the unminified JS being used](https://web.dev/unminified-javascript/) and lists these unminified files and the opportunities to save.
+
+{{ figure_markup(
+  image="unminified-js-audit-scores.png",
+  caption="Percentage distribution of unminified JS audit scores.",
+  description="",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=1572896641&format=interactive",
+  sheets_gid="1539653841",
+  sql_file="lighthouse_unminified_js.sql"
+  )
+}}
+
+It is noted that 67% of mobile pages have an "unminified JS" score between 0.9 and 1.0, meaning that they have a few unminified JS resources.
+
+ This is an improvement of 10% as compared to the results from the last year (2020).
+
+To dive deeper into understanding how many bytes per page are unminified.
+
+{{ figure_markup(
+  image="unminified-js-bytes.png",
+  caption="Percentage distribution of unminified JS bytes per page.",
+  description="",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=1069289665&format=interactive",
+  sheets_gid="1511556407",
+  sql_file="lighthouse_unminified_js_bytes.sql"
+  )
+}}
+
+And it is found that 57% of mobile pages have 0 KB of unminified JS whereas 18% of those have 0-10 kilobytes of unminified JS.
+
+The first party vs. third party analysis in this case shows that 82% of the average mobile page's unminified JS bytes actually come from first-party scripts.
+
+{{ figure_markup(
+  image="average-unminified-js-bytes.png",
+  caption="Average distribution of unminified JS Bytes.",
+  description="",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTpHzC_cMZYj2VLzQ4ODK3uvZkNBXtwdAZriZaBwjLjUM1SGwwmJs9rv8T6OtNdXox29PQ34CasUUwc/pubchart?oid=1119550643&format=interactive",
+  sheets_gid="127709080",
+  sql_file="lighthouse_unminified_js_by_3p.sql"
+  )
+}}
+
+### Source maps
+
+Source maps are files sent along with the JavaScript resource files to let the browser map the minified or compressed file to their original version of that file. This is how source maps help us in debugging in the production environment. 
+
+{{ figure_markup(
+  content="0.09%",
+  caption="Mobile pages using the sourcemap header.",
+  classes="big-number",
+  sheets_gid="1186092564",
+  sql_file="sourcemap_header.sql"
+  )
+}}
+
+It is found that ~98% of mobile pages do not use the SourceMap response header on script resources whereas only 0.09% use it on resources. One reason for this extremely small percentage could be that not many sites choose to put their source code in production through the source map.
```

Comment:
"...put their **original** source code..." instead of "...put their source code..."